lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,1936 @@
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating tagging parameters - " + str(e) +
+                "\n" + traceback.format_exc())
+
+    def validate_tags(self, tags):
+        try:
+            if tags is None:
+                return None
+            list_items = []
+            for tag in tags:
+                key = tag.keys()[0]
+                value = str(tag[key])
+                list_items.append({
+                    'Key': key,
+                    'Value': value
+                })
+            return list_items
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating tags - " + str(e) +
+                "\n" + traceback.format_exc())
+
+    def validate_distribution_config_parameters(
+            self, config, default_root_object, is_ipv6_enabled, http_version,
+            web_acl_id):
+        try:
+            config['default_root_object'] = (default_root_object or '')
+            config['is_i_p_v_6_enabled'] = (is_ipv6_enabled or
+                                            self.__default_is_ipv6_enabled)
+            if http_version is not None:
+                self.validate_attribute_with_allowed_values(
+                    http_version, 'http_version', self.__valid_http_versions)
+                config['http_version'] = http_version
+            if web_acl_id is not None:
+                config['web_a_c_l_id'] = web_acl_id
+            return config
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating distribution config parameters - " +
+                str(e) + "\n" + traceback.format_exc())
+
+    def validate_streaming_distribution_config_parameters(
+            self, config, comment, trusted_signers, s3_origin,
+            default_s3_origin_domain_name,
+            default_s3_origin_origin_access_identity):
+        try:
+            if s3_origin is None:
+                s3_origin = config.get('s3_origin')
+            config['s3_origin'] = self.validate_s3_origin(
+                s3_origin, default_s3_origin_domain_name,
+                default_s3_origin_origin_access_identity)
+            config['trusted_signers'] = self.validate_trusted_signers(
+                trusted_signers)
+            return config
+        except Exception as e:
+            self.module.fail_json(
+                msg="error validating streaming distribution config " +
+                "parameters - " + str(e) + "\n" + traceback.format_exc())
+
+    def validate_common_distribution_parameters(self, config, enabled, aliases,
+                                                logging, price_class, comment,
+                                                is_streaming_distribution):
+        try:
+            if config is None:
+                config = {}
+            if aliases is not None:
+                config['aliases'] = helpers.python_list_to_aws_list(aliases)
+  
```

Comment:
I like that you've put some reusable functions in module_utils.

I'm still reviewing this, but one thing I've noticed is that this module always shows changed=True and e_tag is always modified.
When I run
```
    - name: try one
      cloudfront_distribution:
        state: present
        enabled: false
        e_tag: "E3OUIFFZ9OWBTR"
```
Running that once does modify the expected distribution. But running a second time, a new distribution is created and e_tag disregarded (since with the first run e_tag is modified regardless of whether or not changes should be made). That seems unexpected.